docs/src/pages/components/data-grid/editing/editing.md

```diff
@@ -5,7 +5,123 @@ components: DataGrid, XGrid
+
+{{"demo": "pages/components/data-grid/editing/IsCellEditableGrid.js", "bg": "inline"}}
+
+### Control editing
+
+The `editRowsModel` lets you control the editing.
+
+- implement the `onEditRowModelChange` to control the state of the `GridEditRowsModel` model.
+
+{{"demo": "pages/components/data-grid/editing/EditRowModelControlGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### value Getter
+
+If the column definition process their data using a `valueGetter`.
+
+- set the underlying data by implementing `onEditCellChangeCommitted` event handler invoked when the change is submitted, and the grid is about to update.
+
+{{"demo": "pages/components/data-grid/editing/ValueGetterGrid.js", "bg": "inline"}}
+
+### Validation
+
+To validate cell values on input change.
+
+- set the event handler `onEditCellChange` invoked when a change is triggered by the edit input component.
+- set the event handler `onEditCellChangeCommitted` to validate when the change is committed.
+- or use the `GridEditRowsModel` mentioned in [Control Editing](#control-editing).
+
+{{"demo": "pages/components/data-grid/editing/ValidateRowModelControlGrid.js", "bg": "inline"}}
+
+#### using ApiRef
+
+If you purchased XGrid, it will be quicker to use `apiRef` to do that.
+{{"demo": "pages/components/data-grid/editing/ValidateCellApiRefGrid.js", "bg": "inline"}}
+
+### Server side
+
+Server side validation works just like client side [validation](#validation).
+
+- use the `GridEditRowsModel` mentioned in [Control Editing](#control-editing).
+- or set the event handler `onEditCellChange` for `keydown` validation
+- set the event handler `onEditCellChangeCommitted` to validate and update the server when the change is committed.
+
+**Note:** To prevent the default client side behavior, use `event.stopPropagation()`.
+
+_This demo shows how you can validate a username asynchronously and prevent the user from committing the value while validating._
+{{"demo": "pages/components/data-grid/editing/ValidateServerNameGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Customization
+
+#### Component
+
+To customize the edit component of a column
+
+- set the `renderEditCell` function available in the column definition `GridColDef`.
+
+_The demo lets you edit the ratings by double clicking the cell._
+{{"demo": "pages/components/data-grid/editing/RenderRatingEditCellGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Edit using external button
+
+By default, cells turn to edit mode using the keyboard or by double clicking on it.
+
+- Customize that behaviour and override it completely as shown in this demo.
+
+{{"demo": "pages/components/data-grid/editing/StartEditButtonGrid.js", "bg": "inline"}}
+
+### Events
+
+The editing feature leverages the event capability of the grid and the apiRef.
+The following events can be used to customize the edition:
+
+- `cellEditEnter`: emitted when the cell turns to edit mode.
```

Comment:
Didn't we say that we would be going with documenting the const first instead of the string?

```suggestion
- `GRID_CELL_EDIT_ENTER`: emitted when the cell turns to edit mode.
```

Not sure, I guess @m4theushw planned to work on an RFC to continue the discussion

Reply:
I had the old version on the PR hence why I put the value